06-building-a-program-start-to-finish/writing-tests/output/spec/dplex-test-spec.js

```diff
@@ -0,0 +1,205 @@
+      expect(typeof this.indexInstance.createIndex).toBe('function');
+    });
+
+    it('should check that the class has a readFile method', () => {
+      expect(typeof this.indexInstance.readFile).toBe('function');
+    });
+
+    it('should check that the class has a validateFile method', () => {
+      expect(typeof this.indexInstance.validateFile).toBe('function');
+    });
+
+    it('should check that the class has a tokenize method', () => {
+      expect(typeof this.indexInstance.tokenize).toBe('function');
+    });
+
+    it('should check that the class has a getDocumentTokens method', () => {
+      expect(typeof this.indexInstance.getDocumentTokens).toBe('function');
+    });
+
+    it('should check that the class has a getIndex method', () => {
+      expect(typeof this.indexInstance.getIndex).toBe('function');
+    });
+
+    it('should check that the class has a searchIndex method', () => {
+      expect(typeof this.indexInstance.searchIndex).toBe('function');
+    });
+
+    it('should check that the class has a getSearchResults method', () => {
+      expect(typeof this.indexInstance.getSearchResults).toBe('function');
+    });
+
+    it('should check that the class has a getDocuments method', () => {
+      expect(typeof this.indexInstance.getDocuments).toBe('function');
+    });
+  });
+
+  describe('Create Index', () => {
+    it('should return mapped indices to words in a JSON file', () => {
+      const expectedResult =
+        { alice: [0],
+          falls: [0],
+          into: [0],
+          a: [0, 1, 2],
+          rabbit: [0],
+          hole: [0],
+          and: [0, 1, 2],
+          enters: [0],
+          world: [0],
+          full: [0],
+          of: [0, 1, 2],
+          imagination: [0],
+          in: [0],
+          wonderland: [0],
+          an: [1, 2],
+          unusual: [1, 2],
+          alliance: [1, 2],
+          man: [1, 2],
+          elf: [1, 2],
+          dwarf: [1, 2],
+          wizard: [1, 2],
+          hobbit: [1, 2],
+          seek: [1, 2],
+          to: [1, 2],
+          destroy: [1, 2],
+          powerful: [1, 2],
+          ring: [1, 2],
+          the: [1, 2],
+          lord: [1, 2],
+          rings: [1, 2],
+          fellowship: [1, 2] };
+      this.indexInstance.createIndex(this.books, 'books');
+      expect(this.indexInstance.filesIndexed.books.index)
+        .toEqual(expectedResult);
+    });
+  });
+  describe('Search Index', () => {
+    it('should search through single files that are indexed', () => {
+      const requiredOutput = { alice: [0],
+        and: [0, 1, 2],
+        unusual: [1, 2],
+        imagination: [0] };
+      const searchTerm = this.indexInstance
+        .searchIndex('Alice, and her unusual imagination', 'books');
+      expect(Object.keys(searchTerm[0].indexes))
+        .toEqual(Object.keys(requiredOutput));
+      expect(searchTerm[0].indexes).toEqual(requiredOutput);
+    });
```

Comment:
Is there a test to show when you search for a word not found in the content of indexed file?